tests/features/cached_error_reports.feature

```diff
@@ -11,79 +11,22 @@ Scenario: Sending internal error reports on API <26
     When I run "InternalReportScenario" and relaunch the app
     And I configure the app to run in the "non-crashy" state
     And I run "InternalReportScenario"
-    And I wait to receive a request
-    And the "Bugsnag-Internal-Error" header equals "true"
-    And the payload field "apiKey" is null
-    And the event "context" equals "Crash Report Deserialization"
-    And the event "session" is null
-    And the event "breadcrumbs" is null
-    And the event "app.type" equals "android"
-    And the event "app.duration" is greater than 0
-    And the event "app.durationInForeground" is not null
-    And the event "app.inForeground" is true
-    And the event "device.freeDisk" is greater than 0
-    And the event "device.osName" equals "android"
-    And the event "metaData.BugsnagDiagnostics.filename" is not null
-    And the event "metaData.BugsnagDiagnostics.notifierName" equals "Android Bugsnag Notifier"
-    And the event "metaData.BugsnagDiagnostics.apiKey" equals "ABCDEFGHIJKLMNOPQRSTUVWXYZ012345"
-    And the event "metaData.BugsnagDiagnostics.packageName" equals "com.bugsnag.android.mazerunner"
-    And the event "metaData.BugsnagDiagnostics.notifierVersion" is not null
-    And the event "metaData.BugsnagDiagnostics.fileLength" equals 4
-    And the event "metaData.BugsnagDiagnostics.cacheGroup" is null
-    And the event "metaData.BugsnagDiagnostics.cacheTombstone" is null
+    Then I should receive no requests
 
 @skip_below_android_8
 Scenario: Sending internal error reports on API >=26
     When I run "InternalReportScenario" and relaunch the app
     And I configure the app to run in the "non-crashy" state
     And I run "InternalReportScenario"
-    And I wait to receive a request
-    And the "Bugsnag-Internal-Error" header equals "true"
-    And the payload field "apiKey" is null
-    And the event "context" equals "Crash Report Deserialization"
-    And the event "session" is null
-    And the event "breadcrumbs" is null
-    And the event "app.type" equals "android"
-    And the event "app.duration" is greater than 0
-    And the event "app.durationInForeground" is not null
-    And the event "app.inForeground" is true
-    And the event "device.freeDisk" is greater than 0
-    And the event "device.osName" equals "android"
-    And the event "metaData.BugsnagDiagnostics.filename" is not null
-    And the event "metaData.BugsnagDiagnostics.notifierName" equals "Android Bugsnag Notifier"
-    And the event "metaData.BugsnagDiagnostics.apiKey" equals "ABCDEFGHIJKLMNOPQRSTUVWXYZ012345"
-    And the event "metaData.BugsnagDiagnostics.packageName" equals "com.bugsnag.android.mazerunner"
-    And the event "metaData.BugsnagDiagnostics.notifierVersion" is not null
-    And the event "metaData.BugsnagDiagnostics.fileLength" equals 4
-    And the event "metaData.BugsnagDiagnostics.cacheGroup" is false
-    And the event "metaData.BugsnagDiagnostics.cacheTombstone" is false
+    Then I should receive no requests
```

Comment:
I dont get these scenario changes. Reading these tests its not at all clear. What are they testing?

Reply:
These scenarios used to verify that when sending an invalid JSON file, the notifier sent an incomplete report to our own internal instance. Because we no longer need to deserialize cached JSON files the payload will be dropped by the payload instead - so the scenario has been updated to verify that 0 valid requests are received by mazerunner.